xbmc/windowing/gbm/GBMUtils.cpp

```diff
@@ -21,18 +21,48 @@
 #include "GBMUtils.h"
 #include "utils/log.h"
 
-bool CGBMUtils::InitGbm(struct gbm *gbm, int hdisplay, int vdisplay)
+bool CGBMUtils::CreateDevice(struct gbm *gbm, int fd)
 {
-  gbm->width = hdisplay;
-  gbm->height = vdisplay;
+  if (gbm->device)
+    CLog::Log(LOGWARNING, "CGBMUtils::%s - device already created", __FUNCTION__);
```

Comment:
We should early return in this case.
If early return is wrong, we should at least `gbm_device_destroy(gbm->device);` before we call create again, otherwise this could leak memory.

Reply:
@Kwiboo did you run into a case where gbm->device was already created?

Reply:
Not really, but it looked like create surface or lock front buffer could have been called without destroy surface or release buffer, so added logging for all bad gbm calls in my WIP commit to help spot any issues :)